src/signals/IncidentMap/components/GPSLocation/GPSLocation.test.tsx

```diff
@@ -0,0 +1,151 @@
+/* SPDX-License-Identifier: MPL-2.0 */
+/* Copyright (C) 2022 Gemeente Amsterdam */
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import type { Map } from 'leaflet'
+
+import { DEFAULT_ZOOM } from '../../../../components/AreaMap/AreaMap'
+import configuration from '../../../../shared/services/configuration/configuration'
+import { GPSLocation } from './GPSLocation'
+import type { Props } from './GPSLocation'
+
+jest.mock('@amsterdam/react-maps', () => ({
+  __esModule: true,
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  ...jest.requireActual('@amsterdam/react-maps')!,
+  Marker: ({ ...Props }) => {
+    return <div {...Props}></div>
+  },
+}))
+
+const mockMap = {
+  flyTo: jest.fn(),
+} as unknown as Map
+
+const defaultProps: Props = {
+  map: mockMap,
+  setNotification: jest.fn(),
+}
+
+describe('GPSLocation', () => {
+  const coords = {
+    accuracy: 1234,
+    latitude: 52.3731081,
+    longitude: 4.8932945,
+  }
+
+  const mockGeolocation = {
+    getCurrentPosition: jest.fn().mockImplementation((success) =>
+      Promise.resolve(
+        success({
+          coords,
+        })
+      )
+    ),
+  }
+
+  it('renders the component', () => {
+    render(<GPSLocation {...defaultProps} />)
+
+    expect(
+      screen.getByRole('button', { name: 'Huidige locatie' })
+    ).toBeInTheDocument()
+  })
+
+  it('flies to the location on click', () => {
+    Object.defineProperty(global.navigator, 'geolocation', {
```

Comment:
I  made a small change here. By using object define property we avoid the previous error, but the nice thing is we can remove the ts ignore comments and keep the same result.